GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -991,7 +998,7 @@ private void ParseIndexThreadMain()
 
                     while (this.context.Repository.GVFSLock.GetExternalHolder() != null)

```

Comment:
Another comment related to `parseIndexRequested`.  Do we still need to check `GetExternalHolder`?  Or could this block:

```
while (this.context.Repository.GVFSLock.GetExternalHolder() != null)
{
   if (this.parseIndexRequested.Wait(ParseIndexRequestTimeoutMs))
    {
        break;
    }

    if (this.isStopping)
    {
        return;
    }
}
```

Be replaced with:

```
    if (this.isStopping)
    {
        return;
    }
```